tests/ttnn/unit_tests/operations/ccl/test_barrier_t3000_frequent.py

```diff
@@ -0,0 +1,362 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+from loguru import logger
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from models.utility_functions import skip_for_grayskull
+from tests.ttnn.unit_tests.operations.ccl.test_all_gather import is_unsupported_case_t3k
+
+
+def run_normal(
+    device,
+    num_devices,
+    input_shape,
+    dim,
+    input_dtype,
+    mem_config,
+    layout,
+    num_iters,
+    all_gather_topology,
+    enable_async,
+):
+    print("Running barrier test")
+    if num_iters < 1:
+        pytest.fail("num_iters must be >= 1")
+
+    if enable_async:
+        logger.info(f"Using Async Mode for Barrier Op Dispatch")
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+    input_tensor = torch.rand(input_shape).bfloat16()
+
+    input_tensors = torch.chunk(input_tensor, num_devices, dim)
+    tt_input_tensors = []
+    # Use Async mode based on test input config
+    device.enable_async(enable_async)
+    for i, t in enumerate(input_tensors):
+        tt_input_tensors.append(ttnn.Tensor(t, input_dtype).to(layout).to(device.get_devices()[i], mem_config))
```

Comment:
see my other comment

Reply:
Applied suggestion to all instances of the above